src/component/calculator/Calculator.js

```diff
@@ -0,0 +1,49 @@
+import React, { Component } from 'react';
+import calculate from '../logic/calculate';
+import './Calculator.css';
+
+const operators = ['÷', 'x', '-', '+', '='];
+const digits = ['AC', '+/-', '%', '7', '8', '9', '4', '5', '6', '1', '2', '3', '0', '.'];
+
+class Calculator extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      next: '',
+      total: '',
+      operation: '',
+    };
+  }
+
+  handleClick = (btn) => {
+    this.setState((state) => calculate(
+      {
+        next: state.next,
+        total: state.total,
+        operation: state.operation,
+      },
+      btn,
+    ));
+  }
+
+  render() {
+    const { total, next } = this.state;
+    return (
+      <ul className="calc-body">
+        <li className="calc-screen">
+          {next || total || '0'}
+        </li>
+        <li className="calc-keyboard">
+          <div className="calc-digits">
+            {digits.map((btn) => <button onClick={() => this.handleClick(btn)} type="button" key={btn}>{ btn }</button>) }
+          </div>
+          <div className="calc-operators">
+            {operators.map((btn) => <button onClick={() => this.handleClick(btn)} type="button" key={btn}>{ btn }</button>) }
+          </div>
```

Comment:
- [ ] Great work. Your work is truly impressive and clean. There is a little change you need to make. Remember one of the essences of learning `events`, `useState` and `props` is to be able to practice what you have learned and recognize when to use it. Great work implementing events on JSX and using the component's state 👏. However, you didn't use props. You need to pass data to a child component of the `Calculator` component to be able to use `props`. I would recommend that you create a `Button` component as a child component of the `Calculator` component and pass the `digits` and `operators` as `props` to the `Button` component. 👍 😎